pkg/kubectl/cmd/completion.go

```diff
@@ -126,20 +123,32 @@ func RunCompletion(out io.Writer, boilerPlate string, cmd *cobra.Command, args [
 		return cmdutil.UsageErrorf(cmd, "Unsupported shell type %q.", args[0])
 	}
 
+	return run(out, boilerPlate, cmd.Parent())
+}
+
+func runCompletionBash(out io.Writer, boilerPlate string, kubectl *cobra.Command) error {
 	if len(boilerPlate) == 0 {
 		boilerPlate = defaultBoilerPlate
 	}
 	if _, err := out.Write([]byte(boilerPlate)); err != nil {
 		return err
 	}
-	return run(out, cmd.Parent())
-}
 
-func runCompletionBash(out io.Writer, kubectl *cobra.Command) error {
 	return kubectl.GenBashCompletion(out)
 }
 
-func runCompletionZsh(out io.Writer, kubectl *cobra.Command) error {
+func runCompletionZsh(out io.Writer, boilerPlate string, kubectl *cobra.Command) error {
+	zsh_head := `#compdef kubectl
+`
```

Comment:
Why not replace this line with this?

```go
zshHead := "#compdef kubectl\n"
```

Easier to read and won't annoy the linter.